### PR TITLE
More permissive lower bound for resourcet

### PR DESCRIPTION
**Pull request by jhrcek**
1.3.0 lowerbound for resourcet unnecessarily excludes versions with which the package is working fine (e.g. amazonka even has `resourcet >=1.1`).

We're using stack  lts-20.26 (based on ghc-9.2.8) which has resourcet-1.2.6.
When I add resourcet-1.3.0 to extra-deps in stack.yaml, I get bunch of other version conflicts (e.g. servant-server-0.19.2 from stack lts permits `resourcet >=1.2.2 && <1.3`

Long story short, let's please lower resourcet bound before releasing.

### CODE REVIEW

**Comment by jhrcek** [comment]
Upstream PR https://github.com/axman6/amazonka-s3-streaming/pull/40
